security-analysis/security-analysis-api/src/main/java/com/powsybl/security/detectors/AbstractLimitViolationDetector.java

```diff
@@ -47,6 +52,18 @@ public void checkVoltage(Contingency contingency, Bus bus, Consumer<LimitViolati
         checkVoltage(contingency, bus, bus.getV(), consumer);
     }
 
+    @Override
+    public void checkVoltageAngle(Contingency contingency, VoltageAngleLimit voltageAngleLimit, Consumer<LimitViolation> consumer) {
+        Bus referenceBus = voltageAngleLimit.getTerminalFrom().getBusView().getBus();
+        Bus otherBus = voltageAngleLimit.getTerminalTo().getBusView().getBus();
+        if (referenceBus != null && otherBus != null
+            && referenceBus.getConnectedComponent().equals(otherBus.getConnectedComponent())) {
```

Comment:
I don't think we can compare connected components like this but instead we should use the number